plugins/inputs/postgresql_extensible/postgresql_extensible.go

```diff
@@ -168,22 +168,22 @@ func (p *Postgresql) Gather(acc telegraf.Accumulator) error {
 		sql_query += query_addon
 
 		if p.Query[i].Version <= db_version {
-			rows, err := db.Query(sql_query)
+			rows, err := p.DB.Query(sql_query)
```

Comment:
We should put in a retry handler. Otherwise if the connection gets killed, our query will fail. And if the server has something like an idle connection killer that's shorter than telegraf's gather interval, every single gather will fail.